### PR TITLE
sqlserver: backup import and export to the external storage

### PR DESCRIPTION
**Pull request by mialinx**
### SQLServer

Export/import backups to/from external storage.

Backups blobs are stored as a set of blocks with index in default storage.
During export to external storage, backup blobs are serialized to single .bak files, suitable for downloading and manual restoring using `RESTORE DATABASE FROM DISK='filename.bak'`.
During import from external storage, backup blobs are sliced into the set of blocks appropriate to be restored with wal-g.

### CODE REVIEW

File: internal/databases/sqlserver/blob/server.go [hunk 1]
```diff
@@ -603,36 +604,52 @@ func (bs *Server) HandleBlobPut(w http.ResponseWriter, req *http.Request) {
...
+					return
+				}
+			}
+			if n == 0 {
+				continue
+			}
+			id := fmt.Sprintf("data_%05d", i)
+			name := idx.PutBlock(id, uint64(n))
+			encryptedReader := internal.CompressAndEncrypt(bytes.NewReader(buf[:n]), bs.compressor, bs.crypter)
+			err = folder.PutObject(name, encryptedReader)
```
Comment (ostinru) [inline internal/databases/sqlserver/blob/server.go:630]: I am just curious about why do you split blob into 4Mb chunks.

File: internal/databases/sqlserver/backup_export_handler.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	tracelog.ErrorLogger.FatalOnError(err)
+	defer lock.Close()
+
+	var backupName string
+	var sentinel *SentinelDto
+	backup, err := internal.GetBackupByName(internal.LatestString, utility.BaseBackupPath, folder)
+	tracelog.ErrorLogger.FatalfOnError("can't find latest backup: %v", err)
+	backupName = backup.Name
+	sentinel = new(SentinelDto)
+	err = backup.FetchSentinel(&sentinel)
```
[the remark this answers is not in the document]
Comment (mialinx) [reply internal/databases/sqlserver/backup_export_handler.go:54]: I don't.  But it works too: https://go.dev/play/p/_yTrVgnKpXq

File: internal/databases/sqlserver/blob/server.go (hunk 1 above)
Comment (mialinx) [reply internal/databases/sqlserver/blob/server.go]: SQLServer uses 4Mb blocks by itself. We mimic.

File: internal/databases/sqlserver/blob/server_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package blob
+
+/*
```
Comment (serprex) [inline internal/databases/sqlserver/blob/server_test.go:3]: Why commented?

Comment (mialinx) [reply internal/databases/sqlserver/blob/server_test.go]: Sorry, an unnecessary file was added to commit.  Removed it.

**Review by ostinru — APPROVED** [review]
LGTM

**Pull request merged**